nva-datamodel-java/src/test/java/no/unit/nva/OntologyTest.java

```diff
@@ -0,0 +1,159 @@
+import org.apache.jena.rdf.model.Resource;
+import org.apache.jena.rdf.model.SimpleSelector;
+import org.apache.jena.rdf.model.Statement;
+import org.apache.jena.riot.Lang;
+import org.apache.jena.riot.RDFDataMgr;
+import org.apache.jena.vocabulary.RDF;
+import org.apache.jena.vocabulary.RDFS;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static nva.commons.core.attempt.Try.attempt;
+import static nva.commons.core.ioutils.IoUtils.inputStreamFromResources;
+import static nva.commons.core.ioutils.IoUtils.stringFromResources;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.hamcrest.core.IsIterableContaining.hasItems;
+
+class OntologyTest {
+
+    public static final ObjectMapper MAPPER = JsonUtils.dtoObjectMapper;
+    public static final JsonNode JSON_LD_CONTEXT =
+            attempt(() -> MAPPER.readTree(inputStreamFromResources("publicationContext.json"))).orElseThrow();
+    public static final String ONTOLOGY_STRING = stringFromResources(Path.of("publication-ontology.ttl"));
+    public static final SimpleSelector ANY_CLASS_SELECTOR = new SimpleSelector(null, RDF.type, (RDFNode) null);
+    public static final SimpleSelector ANY_STATEMENT_SELECTOR = new SimpleSelector(null, null, (RDFNode) null);
+    public static final SimpleSelector ONTOLOGY_CLASS_SELECTOR = new SimpleSelector(null, RDF.type, RDFS.Class);
+    public static final SimpleSelector ONTOLOGY_PROPERTY_SELECTOR = new SimpleSelector(null, RDF.type, RDF.Property);
+
+    public static Stream<Class<?>> publicationInstanceProvider() {
+        return PublicationInstanceBuilder.listPublicationInstanceTypes().stream();
+    }
+
+    @Test
+    void shouldContainDistinctDescriptions() {
+        var ontologyValues = ONTOLOGY_STRING.lines()
+                .filter(line -> line.startsWith("nva:"))
+                .collect(Collectors.toList());
+        var distinctValues = ontologyValues.stream().distinct().collect(Collectors.toList());
+        String duplicatesMessage = ontologyValues.equals(distinctValues) ? null : getDuplicatesMessage(ontologyValues);
+        assertThat(duplicatesMessage, distinctValues, is(equalTo(ontologyValues)));
+    }
+
+    @Test
+    void shouldContainEveryVisibleClassOfModel() {
+        var ontologyClasses = extractClassesFromOntology();
+        var modelClasses = new ArrayList<>(getModelClasses()).toArray(String[]::new);
+        assertThat(ontologyClasses, hasItems(modelClasses));
+    }
+
+    @Test
+    void shouldContainEveryVisiblePropertyOfModel() {
+        var ontologyClasses = extractPropertiesFromOntology();
```

Comment:
Names instead ontologyProperties and modelProperties? Not important though

Reply:
Good catch